<a href="https://colab.research.google.com/github/anand8071/ExcelR-classroom/blob/main/Assignments_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd 
import numpy as np
from scipy.stats import norm
import warnings
warnings.filterwarnings("ignore")

 Q7) Calculate Mean, Median, Mode, Variance, Standard Deviation, Range &     comment about the values / draw inferences, for the given dataset
-	For Points,Score,Weigh>
Find Mean, Median, Mode, Variance, Standard Deviation, and Range and also Comment about the values/ Draw some inferences.
Use Q7.csv file


In [2]:
data=pd.read_csv('/content/Q7.csv')

In [3]:
data.head()

Unnamed: 0  Points  Score  Weigh
0          Mazda RX4    3.90  2.620  16.46
1      Mazda RX4 Wag    3.90  2.875  17.02
2         Datsun 710    3.85  2.320  18.61
3     Hornet 4 Drive    3.08  3.215  19.44
4  Hornet Sportabout    3.15  3.440  17.02

In [4]:
data.describe()

Points      Score      Weigh
count  32.000000  32.000000  32.000000
mean    3.596563   3.217250  17.848750
std     0.534679   0.978457   1.786943
min     2.760000   1.513000  14.500000
25%     3.080000   2.581250  16.892500
50%     3.695000   3.325000  17.710000
75%     3.920000   3.610000  18.900000
max     4.930000   5.424000  22.900000

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32 entries, 0 to 31
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  32 non-null     object 
 1   Points      32 non-null     float64
 2   Score       32 non-null     float64
 3   Weigh       32 non-null     float64
dtypes: float64(3), object(1)
memory usage: 1.1+ KB


In [6]:
data.median()


Points     3.695
Score      3.325
Weigh     17.710
dtype: float64

In [7]:
data.mode()

Unnamed: 0  Points  Score  Weigh
0           AMC Javelin    3.07   3.44  17.02
1    Cadillac Fleetwood    3.92    NaN  18.90
2            Camaro Z28     NaN    NaN    NaN
3     Chrysler Imperial     NaN    NaN    NaN
4            Datsun 710     NaN    NaN    NaN
5      Dodge Challenger     NaN    NaN    NaN
6            Duster 360     NaN    NaN    NaN
7          Ferrari Dino     NaN    NaN    NaN
8              Fiat 128     NaN    NaN    NaN
9             Fiat X1-9     NaN    NaN    NaN
10       Ford Pantera L     NaN    NaN    NaN
11          Honda Civic     NaN    NaN    NaN
12       Hornet 4 Drive     NaN    NaN    NaN
13    Hornet Sportabout     NaN    NaN    NaN
14  Lincoln Continental     NaN    NaN    NaN
15         Lotus Europa     NaN    NaN    NaN
16        Maserati Bora     NaN    NaN    NaN
17            Mazda RX4     NaN    NaN    NaN
18        Mazda RX4 Wag     NaN    NaN    NaN
19             Merc 230     NaN    NaN    NaN
20            Merc 240D     NaN    NaN    NaN
21             Merc 280     NaN    NaN    NaN
22            Merc 280C     NaN    NaN    NaN
23           Merc 450SE     NaN    NaN    NaN
24           Merc 450SL     NaN    NaN    NaN
25          Merc 450SLC     NaN    NaN    NaN
26     Pontiac Firebird     NaN    NaN    NaN
27        Porsche 914-2     NaN    NaN    NaN
28       Toyota Corolla     NaN    NaN    NaN
29        Toyota Corona     NaN    NaN    NaN
30              Valiant     NaN    NaN    NaN
31           Volvo 142E     NaN    NaN    NaN

In [8]:
data.var()



Points    0.285881
Score     0.957379
Weigh     3.193166
dtype: float64

## Comment about the values and  some inferences.



**POINTS**

Mean       3.596

Median     3.698

Mode       NaN

Variance   0.285

Range      2.17

Standard Deviation    0.538


1.Mean and Median are Close together.

2.Low standard Deviation and Variance is shows that most of our data is near to mean value.

3.Range is 2.17 mean some variability in data but not extremeley large


**SCORE**


Mean       3.217

Median     3.325

Mode       3.44

Variance   0.957

Range      3.911

Standard Deviation    0.978

1. The mean and median are relatively close together
2. Variance and Standard Deviation are also very small to most of data is near to mean 
3. compare to Points data it is more spread but no extremely large spread because range is 3.911

**WEIGH**

Mean       17.849

Median     17.710

Mode       NaN

Variance   3.18

Range      8.4

Standard Deviation    1.787

1. The mean and median are relatively close together
2. Stadard Deviation and Variance values are very large so data is spread out around the mean
3.Range is 8.4 means data is more spread out around the mean 

# **Q9)**Calculate Skewness, Kurtosis & draw inferences on the following data
#      Cars speed and distance 
# **Q9_a.csv**


In [9]:
import scipy.stats as stats
dataset=pd.read_csv('/content/Q9_a.csv')

In [10]:
dataset.head()

Index  speed  dist
0      1      4     2
1      2      4    10
2      3      7     4
3      4      7    22
4      5      8    16

In [11]:
dataset.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   Index   50 non-null     int64
 1   speed   50 non-null     int64
 2   dist    50 non-null     int64
dtypes: int64(3)
memory usage: 1.3 KB


In [12]:
dataset.describe()

Index      speed        dist
count  50.00000  50.000000   50.000000
mean   25.50000  15.400000   42.980000
std    14.57738   5.287644   25.769377
min     1.00000   4.000000    2.000000
25%    13.25000  12.000000   26.000000
50%    25.50000  15.000000   36.000000
75%    37.75000  19.000000   56.000000
max    50.00000  25.000000  120.000000

In [13]:
skewness=dataset.skew()
skewness

Index    0.000000
speed   -0.117510
dist     0.806895
dtype: float64

**Inferences on the following data Cars speed and distance** 

1.Speed is negetive means data is skewed to the left side,cars have higher speed and  degree of skewmess is very small

2.Distance is positive means data is skewed towards right side and have value is very small so it means cars have shorter stopping points

In [14]:
kurtosis=dataset.kurtosis()
kurtosis

Index   -1.200000
speed   -0.508994
dist     0.405053
dtype: float64

**Inferences on the following data Cars speed and distance**
1. Speed kurtosis is negative so it means it have very small tails 
2. Distance kurtosis is positive so it means it have higher tails or peak.

# **Q9_b**


In [15]:
df= pd.read_csv('/content/Q9_b.csv')        


In [16]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 81 entries, 0 to 80
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  81 non-null     int64  
 1   SP          81 non-null     float64
 2   WT          81 non-null     float64
dtypes: float64(2), int64(1)
memory usage: 2.0 KB


In [17]:
df.head()              


Unnamed: 0          SP         WT
0           1  104.185353  28.762059
1           2  105.461264  30.466833
2           3  105.461264  30.193597
3           4  113.461264  30.632114
4           5  104.461264  29.889149

In [18]:
df.describe()

Unnamed: 0          SP         WT
count   81.000000   81.000000  81.000000
mean    41.000000  121.540272  32.412577
std     23.526581   14.181432   7.492813
min      1.000000   99.564907  15.712859
25%     21.000000  113.829145  29.591768
50%     41.000000  118.208698  32.734518
75%     61.000000  126.404312  37.392524
max     81.000000  169.598513  52.997752

In [19]:
Q9B_skewness=df.skew()
Q9B_skewness

Unnamed: 0    0.000000
SP            1.611450
WT           -0.614753
dtype: float64

In [20]:
Q9B_kurtosis=df.kurtosis()
Q9B_kurtosis

Unnamed: 0   -1.200000
SP            2.977329
WT            0.950291
dtype: float64

# Q12)  Below are the scores obtained by a student in tests 
**34,36,36,38,38,39,39,40,40,41,41,41,41,42,42,45,49,56**

**1)	Find mean, median, variance, standard deviation.**


In [21]:
num=[34,36,36,38,38,39,39,40,40,41,41,41,41,42,42,45,49,56]
num

[34, 36, 36, 38, 38, 39, 39, 40, 40, 41, 41, 41, 41, 42, 42, 45, 49, 56]

In [22]:
type(num)

list

In [30]:
mean=np.mean(num)
median=np.median(num)
mode=stats.mode(num)[0][0]
var=np.var(num)


In [35]:
print("Mean: -", mean)
print("Median: -", median)
print("Mode: -", mode)
print("variance:-", var)
print("standard Deviation: - 4.91")

Mean: - 41.0
Median: - 40.5
Mode: - 41
variance:- 24.11111111111111
standard Deviation: - 4.91


#Q 20) Calculate probability from the given dataset for the below cases

Data _set: Cars.csv
Calculate the probability of MPG  of Cars for the below cases.

MPG <- Cars$MPG

a.	P(MPG>38)

b.	P(MPG<40)

c.    P (20<MPG<50)


In [52]:
cars=pd.read_csv("/content/Cars.csv")

In [53]:
cars.head()

HP        MPG  VOL          SP         WT
0  49  53.700681   89  104.185353  28.762059
1  55  50.013401   92  105.461264  30.466833
2  55  50.013401   92  105.461264  30.193597
3  70  45.696322   92  113.461264  30.632114
4  53  50.504232   92  104.461264  29.889149

In [54]:
cars.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 81 entries, 0 to 80
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   HP      81 non-null     int64  
 1   MPG     81 non-null     float64
 2   VOL     81 non-null     int64  
 3   SP      81 non-null     float64
 4   WT      81 non-null     float64
dtypes: float64(3), int64(2)
memory usage: 3.3 KB


In [55]:
cars.describe()

HP        MPG         VOL          SP         WT
count   81.000000  81.000000   81.000000   81.000000  81.000000
mean   117.469136  34.422076   98.765432  121.540272  32.412577
std     57.113502   9.131445   22.301497   14.181432   7.492813
min     49.000000  12.101263   50.000000   99.564907  15.712859
25%     84.000000  27.856252   89.000000  113.829145  29.591768
50%    100.000000  35.152727  101.000000  118.208698  32.734518
75%    140.000000  39.531633  113.000000  126.404312  37.392524
max    322.000000  53.700681  160.000000  169.598513  52.997752

#a) P(MPG>38)

In [56]:
number_mpg_gt_38=len(cars[cars['MPG']>38])
total_cars=len(cars)

In [57]:
prob_mpg_gt_38= (number_mpg_gt_38)/(total_cars)
prob_mpg_gt_38

0.4074074074074074

#b) P(MPG<40)



In [62]:
number_mpg_lt_40=len(cars[cars['MPG']<40])
total_cars=len(cars)


In [64]:
prob_mpg_lt_40=number_mpg_lt_40/total_cars
prob_mpg_lt_40

0.7530864197530864

#c) P (20<MPG<50)

In [75]:
number_mpg_bet_20_to_50=len(cars[cars['MPG']>20])&len(cars[cars['MPG']<50])
total_cars=len(cars)

In [76]:
prob_mpg_bet_20_to_50= number_mpg_bet_20_to_50/ total_cars
prob_mpg_bet_20_to_50

0.8888888888888888